src/invariant.ts

```diff
@@ -0,0 +1,8 @@
+import { invariant as tsInvariant } from 'ts-invariant';
```

Comment:
Out of curiosity, what was the purpose of the inlining this?

If the purpose was ts 3.7 support, we might want to simultaneously submit that PR against ts-invariant.

Reply:
I agree, this is a temporary sidestep to have the type safety with `ts-invariant` features. As soon as the upstream updates the code this would be removed.

Reply:
Can we add a comment to this for future us so we would know looking at the file why this was done?